From 359842fec0db54e5cd823c689f1a46d39a32c79f Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 15 Aug 2018 14:14:21 +0800
Subject: [PATCH 4609/5242] MLK-19252-3 drm/bridge: sec-dsim: workaround 3
 data lanes case

commit  1ed534501ada2f9b810413b39efd0b6a355c2d41 from
https://source.codeaurora.org/external/imx/linux-imx.git

According to a lot of tests and debug, for the Non-Burst with
Sync Pulse mode with 3 data lanes enable, the DSI peripheral
ADV7535 cannot display correctly, but the output timings seems
to be correct. Until now, the root cause for this issue still
cannot be found. So make this workaround to force to use 2 data
lanes when meeting the 3 lanes requests.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 6e7bc3bfd5b03da698a4024199bb696b792cc14e)
(cherry picked from commit 14f43f0aef26418e10f7840db726e990e557a2d1)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/sec-dsim.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/bridge/sec-dsim.c b/drivers/gpu/drm/bridge/sec-dsim.c
index faa1076..5c3ad03 100644
--- a/drivers/gpu/drm/bridge/sec-dsim.c
+++ b/drivers/gpu/drm/bridge/sec-dsim.c
@@ -367,6 +367,13 @@ static int sec_mipi_dsim_host_attach(struct mipi_dsi_host *host,
 		dsim->panel = panel;
 	}
 
+	/* TODO: DSIM 3 lanes has some display issue, so
+	 * avoid 3 lanes enable, and force data lanes to
+	 * be 2.
+	 */
+	if (dsi->lanes == 3)
+		dsi->lanes = 2;
+
 	dsim->lanes	 = dsi->lanes;
 	dsim->channel	 = dsi->channel;
 	dsim->format	 = dsi->format;
-- 
1.7.9.5

